File: mobile/src/types/tutor.ts
import { TuitionMode } from './common';

export interface TutorSubjectRequest {
  subject_id: string;
  education_level_id: string;
}

export interface TutorProfileUpdateRequest {
  bio?: string;
  mode_of_tuition?: TuitionMode;
  country_id?: string;
  region_id?: string;
  city_id?: string;
  timezone?: string;
  subjects?: TutorSubjectRequest[];
}

export interface PricingUpdateRequest {
  individual_rate?: number;
  group_rate?: number;
  currency?: string;
}

export interface WorkingHoursSlot {
  day_of_week: number;
  start_time: string;
  end_time: string;
  is_working: boolean;
}

export interface WorkingHoursRequest {
  timezone: string;
  slots: WorkingHoursSlot[];
}

export interface TutorSubjectResponse {
  id: string;
  subject_id: string;
  subject_name: string | null;
  category_name: string | null;
  education_level_id: string;
  education_level_name: string | null;
}

export interface WorkingHoursResponse {
  day_of_week: number;
  start_time: string;
  end_time: string;
  is_working: boolean;
  timezone: string;
}

export interface TutorProfileResponse {
  account_id: string;
  first_name: string;
  last_name: string;
  profile_picture_url: string | null;
  bio: string | null;
  mode_of_tuition: TuitionMode | null;
  country_name: string | null;
  region_name: string | null;
  city_name: string | null;
  individual_rate: number | null;
  group_rate: number | null;
  currency: string | null;
  is_profile_complete: boolean;
  timezone: string | null;
  subjects: TutorSubjectResponse[];
  working_hours: WorkingHoursResponse[];
  average_rating: number | null;
  review_count: number;
  sentiment_summary: string | null;
}

export interface TutorCardResponse {
  account_id: string;
  first_name: string;
  last_name: string;
  profile_picture_url: string | null;
  bio: string | null;
  mode_of_tuition: TuitionMode | null;
  city_name: string | null;
  individual_rate: number | null;
  group_rate: number | null;
  currency: string | null;
  subjects: TutorSubjectResponse[];
  average_rating: number | null;
  review_count: number;
}

File: backend/app/schemas/tutor.py
import uuid
from datetime import time
from decimal import Decimal

from pydantic import BaseModel, Field


class TutorSubjectRequest(BaseModel):
    subject_id: uuid.UUID
    education_level_id: uuid.UUID


class TutorProfileUpdateRequest(BaseModel):
    bio: str | None = None
    mode_of_tuition: str | None = Field(None, pattern="^(online|physical|hybrid)$")
    country_id: uuid.UUID | None = None
    region_id: uuid.UUID | None = None
    city_id: uuid.UUID | None = None
    timezone: str | None = None
    subjects: list[TutorSubjectRequest] | None = None


class PricingUpdateRequest(BaseModel):
    individual_rate: Decimal | None = None
    group_rate: Decimal | None = None
    currency: str | None = Field(None, min_length=3, max_length=3)


class WorkingHoursSlot(BaseModel):
    day_of_week: int = Field(ge=0, le=6)
    start_time: time
    end_time: time
    is_working: bool = True


class WorkingHoursRequest(BaseModel):
    timezone: str
    slots: list[WorkingHoursSlot]


class TutorSubjectResponse(BaseModel):
    id: uuid.UUID
    subject_id: uuid.UUID
    subject_name: str | None = None
    category_name: str | None = None
    education_level_id: uuid.UUID
    education_level_name: str | None = None

    model_config = {"from_attributes": True}


class WorkingHoursResponse(BaseModel):
    day_of_week: int
    start_time: time
    end_time: time
    is_working: bool
    timezone: str

    model_config = {"from_attributes": True}


class TutorProfileResponse(BaseModel):
    account_id: uuid.UUID
    first_name: str
    last_name: str
    profile_picture_url: str | None
    bio: str | None
    mode_of_tuition: str | None
    country_name: str | None = None
    region_name: str | None = None
    city_name: str | None = None
    individual_rate: Decimal | None
    group_rate: Decimal | None
    currency: str | None
    is_profile_complete: bool
    timezone: str | None
    subjects: list[TutorSubjectResponse] = []
    working_hours: list[WorkingHoursResponse] = []
    average_rating: float | None = None
    review_count: int = 0
    sentiment_summary: str | None = None

    model_config = {"from_attributes": True}


class TutorCardResponse(BaseModel):
    account_id: uuid.UUID
    first_name: str
    last_name: str
    profile_picture_url: str | None
    bio: str | None
    mode_of_tuition: str | None
    city_name: str | None = None
    individual_rate: Decimal | None
    group_rate: Decimal | None
    currency: str | None
    subjects: list[TutorSubjectResponse] = []
    average_rating: float | None = None
    review_count: int = 0

    model_config = {"from_attributes": True}
